core/src/test/java/com/linecorp/armeria/server/auth/AuthorizerTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.auth;
+
+import static com.spotify.futures.CompletableFutures.exceptionallyCompletedFuture;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+
+import javax.annotation.Nullable;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.netty.channel.DefaultEventLoop;
+import io.netty.channel.EventLoop;
+
+public class AuthorizerTest {
+
+    @Nullable
+    private static EventLoop eventLoop;
+    @Nullable
+    private static ServiceRequestContext serviceCtx;
+
+    @BeforeClass
+    public static void setEventLoop() {
+        eventLoop = new DefaultEventLoop();
+        serviceCtx = mock(ServiceRequestContext.class);
+        when(serviceCtx.contextAwareEventLoop()).thenReturn(eventLoop);
+    }
+
+    @AfterClass
+    public static void clearEventLoop() {
+        serviceCtx = null;
+        if (eventLoop != null) {
+            eventLoop.shutdownGracefully();
+            eventLoop = null;
+        }
+    }
+
+    @Test
+    public void orElseFirst() {
+        final Authorizer<Object> a = newMock();
+        when(a.authorize(any(), any())).thenReturn(completedFuture(true));
+        final Authorizer<Object> b = newMock();
+
+        final Boolean result = a.orElse(b).authorize(serviceCtx, new Object())
+                                .toCompletableFuture().join();
+        assertThat(result).isTrue();
+        verify(a, times(1)).authorize(any(), any());
```

Comment:
We should go ahead and verify the arguments too (might want to use a string instead of object to make it easier).